dashboard/src/main/home/cluster-dashboard/expanded-chart/status/useLogs.ts

```diff
@@ -94,6 +96,51 @@ export const useLogs = (
     } catch (error) {}
   };
 
+  /**
+   * Updates the `logs` for `containerName` with `newLogs`
+   * @param containerName Name of the container
+   * @param newLogs New logs to update for
+   */
+  const updateContainerLogs = (
+    containerName: string,
+    newLogs: Anser.AnserJsonEntry[][]
+  ) => {
+    setLogs((logs) => {
+      const tmpLogs = { ...logs };
+      let containerLogs = tmpLogs[containerName] || [];
+
+      containerLogs.push(...newLogs);
+      // this is technically not as efficient as things could be
+      // if there are performance issues, a deque can be used in place of a list
+      // for storing logs
+      if (containerLogs.length > MAX_LOGS) {
+        containerLogs.slice(MAX_LOGS);
```

Comment:
Hey @meehawk two small items here! Let me know what you think!
 
I think a better approach would be to do:
```suggestion
        const logsToBeRemoved = newLogs.length < LOGS_BUFFER_SIZE ? newLogs.length : LOGS_BUFFER_SIZE;
        containerLogs = containerLogs.slice(logsToBeRemoved);
```

This way we would only be deleting the necessary amount of logs required for the new logs to not pass the  MAX_LOGS size. Although this would leave us on the edge each time, so open to hearing other ideas here! 

Also added on the second line the re-assignation of the containerLogs variable as if we don't do this, the containerLogs will not be updated with the reduced array, [reference docs here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/slice)

Reply:
i think the reason we dont reassign the container array is so that we can still keep the "rendered output" corresponding to the previous logs in the DOM. If we change the reference, we also lose older logs from the UI

Reply:
Hey @jnfrati thanks for the suggestion.

1. Agree with the suggestion. Done!
2. Nice catch! Updated the code